### PR TITLE
feat: move envrc to top level

### PR DESCRIPTION
**Pull request by smartcontracts**

### CODE REVIEW

File: .envrc.example [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+# NOTE: If you want to deploy contracts from scratch you MUST reload this
+#       variable to ensure the salt is regenerated and the contracts are
+#       deployed to new addresses (otherwise deployment will fail)
+export IMPL_SALT=$(openssl rand -hex 32)
+
+# Name for the deployed network
+export DEPLOYMENT_CONTEXT=getting-started
+
+# Optional Tenderly details for simulation link during deployment
+export TENDERLY_PROJECT=
```
Comment (tynes) [inline .envrc.example:47]: I don't believe these are used anymore, worth a quick `grep`

File: .gitignore [hunk 2]
```diff
@@ -30,6 +30,7 @@ packages/contracts-bedrock/deployments/anvil
 .env
 .env*
 !.env.example
```
Comment (tynes) [inline .gitignore:32]: Do we still have a `.env.example`?

File: packages/contracts-bedrock/scripts/getting-started-config.sh [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+# Generate the config file
+config=$(cat << EOL
+{
+  "finalSystemOwner": "$GS_ADMIN_ADDRESS",
+  "portalGuardian": "$GS_ADMIN_ADDRESS",
+
+  "l1StartingBlockTag": "$blockhash",
+
+  "l1ChainID": 5,
```
Comment (tynes) [inline packages/contracts-bedrock/scripts/getting-started-config.sh]: still doing goerli?

File: op-node/p2p/cli/load_signer.go [hunk 4]
```diff
@@ -19,7 +20,7 @@ func LoadSignerSetup(ctx *cli.Context) (p2p.SignerSetup, error) {
 	if key != "" {
 		// Mnemonics are bad because they leak *all* keys when they leak.
 		// Unencrypted keys from file are bad because they are easy to leak (and we are not checking file permissions).
-		priv, err := crypto.HexToECDSA(key)
+		priv, err := crypto.HexToECDSA(strings.TrimPrefix(key, "0x"))
```
Comment (trianglesphere) [inline op-node/p2p/cli/load_signer.go:23]: Nice fix, but seems unrelated to this PR
